src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2304,6 +2305,7 @@ private void ConnectClient(object state)
                 }
                 catch (SocketException) 
                 {
+                    Task.Delay(100).Wait(); // Give the other end a chance to call Accept().
```

Comment:
Do you know what the behavior of `Thread.Yield` is if the running thread is the only thread on its processor that is ready to run? If it doesn't actually yield in that case, it won't be sufficient. I think that a delay is the best of our bad options here.